xml/admin_backup.xml

```diff
@@ -0,0 +1,398 @@
+ </screen>
+    </step>
+    <step>
+     <para>
+      Finally, copy the created files <filename>backup-admin-node-[date].tgz</filename>
+      and <filename>/var/tmp/db-backup-[date].sql.gz</filename> to
+      a location of your choosing; in accordance with your best practices for
+      backup files.
+     </para>
+     <important>
+      <title>Sensitive Data</title>
+      <para>
+       Please note that the data backup files contain senstive information that
+       allow full cluster access and must be handled securely.
+      </para>
+     </important>
+    </step>
+   </procedure>
+  </sect2>
+
+  <sect2 xml:id="sec.admin.backup.admin.restore">
+   <title>Restore Procedure</title>
+   <procedure>
+    <step>
+     <para>
+      Set up a new machine for the &admin_node; by following
+      <xref linkend="sec.deploy.nodes.admin_install" />.
+     </para>
+     <para>
+      Make sure the &admin_node; is registered against &scc; and has the latest
+      packages installed from the update channel.
+     </para>
+     <para>
+      You must be able to access &dashboard; indicating all containers are
+      correctly started. There is, however, no need to complete the
+      configuration steps in &dashboard;.
+     </para>
+    </step>
+    <step>
+     <para>
+      Copy the <filename>backup-admin-node-[date].tgz</filename> and
+      <filename>db-backup-[date].sql.gz</filename> of the most recent backups
+      onto the newly created &admin_node; machine. Make sure to replace
+      <literal>[date]</literal> with the actual date subsituted during the
+      backup procedure.
+     </para>
+    </step>
+    <step>
+     <para>
+      SSH into the new &admin_node; machine and switch to the location where
+      you have stored the backup files.
+     </para>
+    </step>
+    <step>
+     <para>
+      Run the following command to unpack the data backup:
+     </para>
+<screen>&prompt.root;<command>tar xzf <replaceable>backup-admin-node-[date].tgz</replaceable> -C /</command></screen>
+     <para>
+      Run the following command to load the database backup into the MariaDB
+      container. Make sure to replace <literal>[date]</literal> with the actual
+      date subsituted during the backup procedure.
+     </para>
+<screen>&prompt.root;<command>zcat <replaceable>db-backup-[date].sql.gz</replaceable> | \
+docker exec -i $(docker ps -qf name="velum-mariadb") \
+mysql --password=$(cat /var/lib/misc/infra-secrets/mariadb-root-password) velum_production</command></screen>
+     <note>
+      <para>
+       Note: These instructions assume that all cluster member node hostnames
+       are resolvable by one another via DNS.
+      </para>
+     </note>
+    </step>
+    <step>
+     <para>
+      Once the files have been unpacked successfully, reboot the node.
+     </para>
+    <screen>&prompt.root;<command>reboot now</command></screen>
+    </step>
+    <step>
```

Comment:
One more step is needed - users have to re-accept salt minions (either in Velum or by SSH) because we don't do a backup of the salt-master container so we are loosing info about previously accepted salt minions.

Without this step the velum will show up all the nodes under "Nodes" and "Pending Nodes" at the same time with the same machine-id...

Run this command over SSH to re-accept the salt minions before entering Velum and then the issue will be not visible for users (and user impression will be better):
docker exec -i $(docker ps -qf name="salt-master") salt-key --accept-all --yes

Reply:
It appears that we do backup of etc/salt/pki which should contain all salt-key minion data. I will re-validate if this step is really needed once again later.

Reply:
Unfortunately this problem persists.
![reaccept_minions_after_restore](https://user-images.githubusercontent.com/12828077/53647229-606df000-3c3d-11e9-94fa-d1d368eafa91.png)
(Ignore wrong node hostnames, it's known issue - it was there before I performed b&r)

I repeated all the backup & restore steps again by following updated doc https://w3.nue.suse.com/~mnapp/2019-03-01/book.caasp.admin/cha.admin.backup.html#sec.admin.backup

Reply:
I found maybe related problem - admin node changes its `/etc/machine-id` and possibly it may lead to some unpredictable problems (like this???)

Good thing is that salt-master machine-id didn't change and admin node is named 'admin' for salt (minion name) and not `<machine-id>`. In addition entry in /etc/hosts for admin also doesn't contain `<machine-id>` hash.

Anyway is there any way how can we preserve admin's machine-id? What about to add `/etc/machine-id` into backup?

**original node**
```
admin:~ # cat /etc/machine-id 
02a0cffb2fa54152b458ede6a551796c
admin:~ # docker exec -i $(docker ps -f name=salt-master -q) cat /etc/machine-id
7b9462cb98a677069326e3f05b2926d5
```

**restored node**
```
admin:~ # cat /etc/machine-id 
11a0cffb2fa54152b458beefa551deed
admin:~ # docker exec -i $(docker ps -f name=salt-master -q) cat /etc/machine-id
7b9462cb98a677069326e3f05b2926d5
```

Reply:
Let me give it a try.

Reply:
@Klaven any progress here?

Reply:
ah, now I see where is the problem with re-accepting minions. When you deploy a new admin node a current worker nodes are trying contact their salt-master (not restored yet) on the same address as it was before but the admin doesn't know them yet... Anyway --accept-all is ok for me.